### PR TITLE
Add endpoint_url in test_connection

### PR DESCRIPTION
**Pull request by ieunea1128**
---
In this pull request, we are resolving an issue related to the failure of the test connection caused by an incorrect STS URL. With this pull request, I've ensure that if the 'test_endpoint_url' parameter is provided, the STS check will be performed using that specific URL. However, if the parameter is not provided, the default URL will be used instead.

related: #32652

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by uranusjr** [comment]
Please set up pre-commit (see link in the above comment) and fix static check failures.

File: airflow/providers/amazon/aws/hooks/base_aws.py [hunk 1]
```diff
@@ -796,7 +796,10 @@ def test_connection(self):
         """
         try:
             session = self.get_session()
-            conn_info = session.client("sts").get_caller_identity()
+            test_endpoint_url = self.conn_config.extra_config.get('test_endpoint_url') or None
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/hooks/base_aws.py]: By default, `get` returns None if the key does not exist.

```suggestion
            test_endpoint_url = self.conn_config.extra_config.get('test_endpoint_url')
```

File: airflow/providers/amazon/aws/hooks/base_aws.py [hunk 2]
```diff
@@ -796,7 +796,10 @@ def test_connection(self):
...
+            conn_info = session.client("sts",
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/hooks/base_aws.py]: Could you also add a unit test for this new branch

**Review by vincbeck — CHANGES_REQUESTED** [review]

File: airflow/providers/amazon/aws/hooks/base_aws.py [hunk 3]
```diff
@@ -796,7 +796,11 @@ def test_connection(self):
         """
         try:
             session = self.get_session()
-            conn_info = session.client("sts").get_caller_identity()
+            test_endpoint_url = self.conn_config.extra_config.get("test_endpoint_url")
+            conn_info = session.client(
+                "sts",
+                endpoint_url=test_endpoint_url,
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/hooks/base_aws.py:802]: Did you tested it when `test_endpoint_url ` is None. My experience with `boto3` is they make a difference between a parameter value being `None` and a parameter not provided to the function

Comment (ieunea1128) [reply airflow/providers/amazon/aws/hooks/base_aws.py:802]: Both of them returned successful result when I've tested both being None and a parameter not provided to the function.

I've executed code below
```python
conn = Connection(
    conn_id="sample_aws_connection",
    conn_type="aws",
    login="access_key",  # Reference to AWS Access Key ID
    password="secret_key",  # Reference to AWS Secret Access Key
    extra={
        "test_endpoint_url": "https://sts.us-east-1.amazonaws.com",
        "region_name": "us-east-1"
    },
)

env_key = f"AIRFLOW_CONN_{conn.conn_id.upper()}"
conn_uri = conn.get_uri()
os.environ[env_key] = conn_uri
conn.test_connection()
```

and I've compared results of two connection. Both of them returned successful result, and there's difference of ```RequestID``` in ```ResponseMetadata``` and ```x-amzn-requestid```, ```date``` in ```HTTPHeaders``` 
```python
conn_info = session.client(
	"sts",
	endpoint_url=test_endpoint_url,
	).get_caller_identity()

conn_info_without_endpoint_url = session.client(
	"sts"
	).get_caller_identity()
```

Could you let me know if there's anything I've missed?

Comment (ieunea1128) [reply airflow/providers/amazon/aws/hooks/base_aws.py:802]: Oops there's difference in endpoint of sts session. When I've printed endpoint, former returns url of ```test_endpoint_url``` and later returns Global endpoint.

Comment (vincbeck) [reply airflow/providers/amazon/aws/hooks/base_aws.py:802]: In the code you provided, you dont test when `endpoint_url` is `None` but I assume you tested it as you said. LGTM then

**Review by vincbeck — APPROVED** [review]

**Comment by ieunea1128** [comment]
@vincbeck After making the mentioned change, the test for ```test_hook_connection_test_failed``` in ```test/providers/amazon/aws/hooks/test_base_aws.py``` started asserting an error due to the absence of the ```test_endpoint_url``` in ```mock_sts_client```. To resolve this, I will make the necessary modifications. 
By the way, Would it be better to include a unit test specifically for test_endpoint_url to ensure its validity?

**Comment by vincbeck** [comment]
> @vincbeck After making the mentioned change, the test for `test_hook_connection_test_failed` in `test/providers/amazon/aws/hooks/test_base_aws.py` started asserting an error due to the absence of the `test_endpoint_url` in `mock_sts_client`. To resolve this, I will make the necessary modifications. By the way, Would it be better to include a unit test specifically for test_endpoint_url to ensure its validity?

That would be great :)

**Comment by ieunea1128** [comment]
@vincbeck I've added unit test for test_endpoint_url. Would you mind to review my code?

**Review by vincbeck — APPROVED** [review]
LGTM. I'll merge it once the CI is green

**Pull request merged**